packages/appium/lib/cli/setup-command.js

```diff
@@ -0,0 +1,249 @@
+  log.info(`Uninstalling drivers/plugins in APPIUM_HOME[${appiumHome}]`);
+
+  for (const driverName of _.keys(driverConfig.installedExtensions)) {
+    await uninstallDriver(driverName, driverConfig);
+  }
+
+  for (const driverName of _.keys(pluginConfig.installedExtensions)) {
+    await uninstallPlugin(driverName, pluginConfig);
+  }
+}
+
+
+/**
+ * Install drivers listed in DEFAULT_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupDriverMobile(driverConfig) {
+  for (const driverName of getPresetDrivers(SUBCOMMAND_MOBILE)) {
+    await setupDriver(driverName, driverConfig);
+  }
+}
+
+/**
+ * Install all of known drivers listed in BROWSER_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupDriverBrowser(driverConfig) {
+  for (const driver of getPresetDrivers(SUBCOMMAND_BROWSER)) {
+    await setupDriver(driver, driverConfig);
+  }
+}
+
+/**
+ * Install all of known drivers listed in DESKTOP_APP_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ */
+async function setupDriverDesktopApp(driverConfig) {
+  for (const driver of getPresetDrivers(SUBCOMMAND_DESKTOP)) {
+    await setupDriver(driver, driverConfig);
+  }
+}
+
+/**
+ * Install the given driver name. It skips the installation if the given driver name was already installed.
+ * @param {string} driverName
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupDriver(driverName, driverConfig) {
+  if (_.keys(driverConfig.installedExtensions).includes(driverName)) {
+    log.info(`${driverName} (${driverConfig.installedExtensions[driverName].version}) is already installed. ` +
+      `Skipping the installation.`);
+    return;
+  }
+
+  /** @type {Args} */
+  const driverConfigArgs = {
+    'subcommand': 'driver',
+    'driverCommand': 'install',
+    'driver': driverName,
+    'extraArgs': []
+  };
+  await runExtensionCommand(driverConfigArgs, driverConfig);
+}
+
+/**
+ * Uninstall the given driver name.
+ * @param {string} driverName
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function uninstallDriver(driverName, driverConfig) {
+  /** @type {Args} */
+  const driverConfigArgs = {
+    'subcommand': 'driver',
+    'driverCommand': 'uninstall',
+    'driver': driverName,
+    'extraArgs': []
+  };
+  await runExtensionCommand(driverConfigArgs, driverConfig);
+}
+
+/**
+ * Install drivers listed in DEFAULT_PLUGINS.
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+async function setupPluginDefault(pluginConfig) {
+  for (const pluginName of DEFAULT_PLUGINS) {
+    await setupPlugin(pluginName, pluginConfig);
+  }
+}
+
+/**
+ * Install the given plugin name. It skips the installation if the given plugin name was already installed.
+ * @param {string} pluginName
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+async function setupPlugin(pluginName, pluginConfig) {
```

Comment:
this method is very similar to the driver one. Maybe just create one, like `setupExtension` and pass it as an arg whether this is a driver or a plugin to reduce the amount of duplicated code. Same applies to uninstall functions

Reply:
I liked to build the command explicitly, but let me define shared methods if it looks better. Current type does not work well for the below definition case with `@typedef {'driver'|`plugin`} driverOrPlugin` so maybe need to tweak type more though
```
{
    'subcommand': driverOrPlugin,
    [`${driverOrPlugin}Command`]: 'install',
    [`${driverOrPlugin}`]: extName,
    'extraArgs': []
  }
```